pallets/gear/src/tests.rs

```diff
@@ -1369,12 +1369,115 @@ fn wake_messages_after_program_inited() {
     })
 }
 
+#[test]
+fn exit_init() {
+    use tests_exit_init::WASM_BINARY_BLOATY;
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        System::reset_events();
+
+        let code = WASM_BINARY_BLOATY.expect("Wasm binary missing!").to_vec();
+        assert_ok!(GearPallet::<Test>::submit_program(
+            Origin::signed(USER_1).into(),
+            code.clone(),
+            vec![],
+            Vec::new(),
+            10_000_000u64,
+            0u128
+        ));
+
+        let program_id = utils::get_last_program_id();
+
+        run_to_block(2, None);
+
+        assert!(!Gear::is_failed(program_id));
+        assert!(!Gear::is_initialized(program_id));
+
+        let actual_n = Gear::mailbox(USER_1)
+            .map(|t| t.into_values().fold(0usize, |i, _| i + 1))
+            .unwrap_or(0);
+
+        assert_eq!(actual_n, 0);
+
+        // Program is removed and can be submitted again
+        assert_ok!(GearPallet::<Test>::submit_program(
```

Comment:
I think it can be submitted regardless if previous one `exit`-ed?

Reply:
The salt is the same. So if I'm not mistaken ID of the program should be the same and an error should be returned.